streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1094,16 +1100,32 @@ private int getNumStreamThreads(final boolean hasGlobalTopology) {
     }
 
     private int getNextThreadIndex() {
-        final HashSet<String> names = new HashSet<>();
-        processStreamThread(thread -> names.add(thread.getName()));
-        final String baseName = clientId + "-StreamThread-";
-        for (int i = 1; i <= threads.size(); i++) {
-            final String name = baseName + i;
-            if (!names.contains(name)) {
-                return i;
+        final HashSet<String> allLiveThreadNames = new HashSet<>();
+        final AtomicInteger maxThreadId = new AtomicInteger(1);
```

Comment:
Do we really need an atomic integer here? `maxThreadId` is only used in the synchronized block.

Reply:
It's because variables used in a lamdba must "final or effectively final" whatever that means